crates/indexer/src/models/move_modules.rs

```diff
@@ -0,0 +1,138 @@
+    FieldCount,
+    Identifiable,
+    Insertable,
+    Queryable,
+    Serialize,
+)]
+#[belongs_to(Transaction, foreign_key = "transaction_version")]
+#[primary_key(transaction_version, write_set_change_index)]
+#[diesel(table_name = "move_modules")]
+pub struct MoveModule {
+    pub transaction_version: i64,
+    pub write_set_change_index: i64,
+    pub transaction_block_height: i64,
+    pub name: String,
+    pub address: String,
+    pub bytecode: Option<Vec<u8>>,
+    pub exposed_functions: Option<serde_json::Value>,
+    pub friends: Option<serde_json::Value>,
+    pub structs: Option<serde_json::Value>,
+    pub is_deleted: bool,
+    // Default time columns
+    pub inserted_at: chrono::NaiveDateTime,
+}
+
+pub struct MoveModuleByteCodeParsed {
+    pub address: String,
+    pub name: String,
+    pub bytecode: Vec<u8>,
+    pub exposed_functions: serde_json::Value,
+    pub friends: serde_json::Value,
+    pub structs: serde_json::Value,
+}
+
+impl MoveModule {
+    pub fn from_write_module(
+        write_module: &WriteModule,
+        write_set_change_index: i64,
+        transaction_version: i64,
+        transaction_block_height: i64,
+    ) -> Self {
+        let parsed_data = Self::convert_move_module_bytecode(&write_module.data);
+        Self {
+            transaction_version,
+            transaction_block_height,
+            write_set_change_index,
+            name: parsed_data
+                .as_ref()
+                .map(|d| d.name.clone())
+                .unwrap_or_default(),
+            address: write_module.address.to_string(),
+            bytecode: parsed_data.as_ref().map(|d| d.bytecode.clone()),
+            exposed_functions: parsed_data.as_ref().map(|d| d.exposed_functions.clone()),
+            friends: parsed_data.as_ref().map(|d| d.friends.clone()),
+            structs: parsed_data.as_ref().map(|d| d.structs.clone()),
+            is_deleted: false,
+            inserted_at: chrono::Utc::now().naive_utc(),
+        }
+    }
+
+    pub fn from_delete_module(
+        delete_module: &DeleteModule,
+        write_set_change_index: i64,
+        transaction_version: i64,
+        transaction_block_height: i64,
+    ) -> Self {
+        Self {
+            transaction_version,
+            transaction_block_height,
+            write_set_change_index,
+            name: delete_module.module.name.to_string(),
+            address: delete_module.address.to_string(),
+            bytecode: None,
+            exposed_functions: None,
+            friends: None,
+            structs: None,
+            is_deleted: true,
+            inserted_at: chrono::Utc::now().naive_utc(),
+        }
+    }
+
+    pub fn convert_move_module_bytecode(
+        mmb: &MoveModuleBytecode,
+    ) -> Option<MoveModuleByteCodeParsed> {
+        mmb.clone().try_parse_abi().map_or_else(
+            |e| {
+                aptos_logger::warn!(
+                    "[fh-stream] Could not decode MoveModuleBytecode ABI. Error: {:?}",
```

Comment:
`[fh-stream]`? Also, why is this only a warning? How often do this happen? When, why?

Reply:
obviously i copy pasted lmao. do you want to panic? i don't think we should panic. we can easily see this by searching for empty field.

Reply:
I think panic is better than empty fields: when will we ever search? We could have millions of rows missing data and notknow